libraries/core_libs/network/rpc/Test.cpp

```diff
@@ -47,11 +47,17 @@ Json::Value Test::get_dag_block(const Json::Value &param1) {
   try {
     if (auto node = full_node_.lock()) {
       blk_hash_t hash = blk_hash_t(param1["hash"].asString());
-      auto blk = node->getDagBlockManager()->getDagBlock(hash);
-      if (!blk) {
-        res = "Block not available \n";
+      const auto &dag_blk_mgr = node->getDagBlockManager();
+      auto known_block = dag_blk_mgr->isDagBlockKnown(hash);
+      if (known_block) {
+        auto blk = dag_blk_mgr->getDagBlock(hash);
+        if (!blk) {
```

Comment:
I do not understand this change, why not call `getDagBlock` as `isDagBlockKnown` calls inside `getDagBlock` so you basically calling it 2x

Reply:
I am not sure how to handle the rpc calls for light node. In current implementation if in a light node if dag block is in the part of period_data that we deleted isDagBlockKnown will return true but getDagBlock will return a null value because the period_data is missing. I am not sure if this is the best way to go.